pkg/auth/group/group_adder.go

```diff
@@ -0,0 +1,32 @@
+package group
+
+import (
+	"net/http"
+
+	"github.com/openshift/origin/pkg/auth/api"
+	"github.com/openshift/origin/pkg/auth/authenticator"
+)
+
+// GroupAdder wraps a request authenticator, and adds the specified groups to the returned user when authentication succeeds
+type GroupAdder struct {
+	Authenticator authenticator.Request
+	Groups        []string
+}
+
+func (g *GroupAdder) AuthenticateRequest(req *http.Request) (api.UserInfo, bool, error) {
```

Comment:
The other authenticators from pkg/auth/authenticator/interfaces are left out.  Technically they aren't used by the API server, but it feels strange to have potentially incomplete information returned by them.

I don't have an idea that I particularly like that would address this.